tests/plugins/test_variable_redefinition_in_foreach.py

```diff
@@ -0,0 +1,55 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+# SPDX-FileCopyrightText: 2024 Greenbone AG
+
+from pathlib import Path
+
+from tests.plugins import PluginTestCase
+from troubadix.plugin import LinterWarning
+from troubadix.plugins.variable_redefinition_in_foreach import (
+    CheckVariableRedefinitionInForeach,
+)
+
+
+class CheckVariableRedefinitionInForeachTestCase(PluginTestCase):
+    def test_ok(self):
+        nasl_file = Path(__file__).parent / "test.nasl"
+        content = (
+            "urls = ['foo', 'bar']\n"
+            "foreach url ( urls ) {}\n"
+            "url1 = 'foo'\n"
+            "foreach url(make_list(url1,'bar'))"
```

Comment:
```suggestion
            'urls = ["foo", "bar"];\n'
            "foreach url(urls) {}\n"
            'url1 = "foo";\n'
            'foreach url(make_list(url1, "bar")) {}\n'
```

To "transform" those to valid NASL code / use the most common coding style.

Maybe we could also extend them a little like e.g.:

```suggestion
            'urls = ["foo", "bar"];\n'
            "foreach url(urls) {\n  display(url);\n}\n"
            'url1 = "foo";\n'
            'foreach url(make_list(url1, "bar")) {\n  display(url);\n}'
```

Reply:
implemented in 315a91465f4fde6a0c1a597f8e8d98e22b221062